tests/cephadm/test_bootstrap.py

```diff
@@ -128,18 +129,37 @@ def parse(string):
     data = json.dumps({"password": passwd_, "username": user_})
     url_ = f"{host}/api/auth" if not host.endswith("/") else f"{host}api/auth"
 
-    session = requests.Session()
-    session.headers = {
-        "accept": "application/vnd.ceph.api.v1.0+json",
-        "content-type": "application/json",
-    }
+    def dashboard_api_status():
+        try:
+            session = requests.Session()
+            session.headers = {
+                "accept": "application/vnd.ceph.api.v1.0+json",
+                "content-type": "application/json",
+            }
+
+            resp = session.post(url_, data=data, verify=False)
+            if not resp.ok:
+                raise BootStrapValidationFailure(
+                    f"Status code: {resp.status_code}\nResponse: {resp.text}"
+                )
+            log.info(
+                "Dashboard API login is successful, status code : %s" % resp.status_code
+            )
+            return True
+        except requests.exceptions.ConnectionError:
+            log.info("connection timeout.. try again")
+        return False
```

Comment:
IMO, we should avoid nesting this method. 

Also, let's rename the method to `verify_dashboard_login`